lib/elements/text.js

```diff
@@ -154,27 +155,68 @@ class TextPrompt extends Prompt {
 
   render() {
     if (this.closed) return;
-    if (!this.firstRender) {
-      if (this.outputError)
-        this.out.write(cursor.down(lines(this.outputError) - 1) + clear(this.outputError));
-      this.out.write(clear(this.outputText));
+    if (this.firstRender) {
+      this.out.write('\n' + cursor.up(1));
+      this.prevRowCount = 0;
     }
     super.render();
-    this.outputError = '';
+    let outputPostMessage = ``;
+    this.postMessage = ``;
 
-    this.outputText = [
+    this.outputPrompt = [
       style.symbol(this.done, this.aborted),
       color.bold(this.msg),
       style.delimiter(this.done),
-      this.red ? color.red(this.rendered) : this.rendered
+      ``
     ].join(` `);
-
-    if (this.error) {
-      this.outputError += this.errorMsg.split(`\n`)
-          .reduce((a, l, i) => a + `\n${i ? ' ' : figures.pointerSmall} ${color.red().italic(l)}`, ``);
+    this.outputText = this.red ? color.red(this.rendered) : this.rendered;
+
+    const outputLength = (
+      /* symbol length */
+      2 +
+      /* prompt */
+      this.msg.length +
+      /* delimiter */
+      3 +
+      /* input */
+      this.value.length
+    )
+
+    const rowCount = lines(` `.repeat(outputLength)) - 1;
+    const curRowLength = outputLength - (this.out.columns * rowCount);
+    const cursorGrossPos = this.cursor - this.value.length + outputLength;
+    const cursorRowCnt = lines(` `.repeat(cursorGrossPos)) - 1;
+    const cursorHorzPos = cursorGrossPos - (this.out.columns * cursorRowCnt);
+    const cursorRowOffset = rowCount - cursorRowCnt;
```

Comment:
This calculates the current position of the cursor accurately.  It handles long, wrapping inputs, wrapping hints/ errors, cursor forward, backward.